### ISSUE TITLE
Support listing files in a directory Path

### ISSUE THREAD

**Issue by jeffdgr8** [Kotlin/kotlinx-io#222]
Listing files within a directory is one of the few file APIs my library tests require that `kotlinx-io` doesn't currently support.

Similar API references:

* [JVM File.listFiles](https://docs.oracle.com/javase/8/docs/api/java/io/File.html#listFiles--)
* [Apple Foundation NSFileManager.contentsOfDirectoryAtPath:error:](https://developer.apple.com/documentation/foundation/nsfilemanager/1414584-contentsofdirectoryatpath)
* [Okio FileSystem.list](https://square.github.io/okio/3.x/okio/okio/okio/-file-system/list.html)

**Comment by jeffdgr8** [comment]
My current use case in tests is fairly simple. I'm using the API to either check the number of files in a directory using the size of the file list, or to recursively delete all the contents of a directory.

The file/path list API works well for both of these requirements. An iterable sequence could also work, although it requires iterating and terminating the sequence to get the size. I would think if the underlying platform file API being used is already returning a list or array, there's probably not much value in mapping that existing collection to a sequence.

The other API options could also be useful. It might make sense to offer these filter/iterable APIs in addition to a simple file/path list API. A recursive iteration API would be especially useful for the recursive delete operation.

**Comment by StefanOltmann** [comment]
I think just returning a list of strings with the file names in the directory like `java.io.File.list()` does is the easiest way to get this started.

**Comment by lppedd** [comment]
> I think just returning a list of strings with the file names in the directory like java.io.File.list() does is the easiest way to get this started.

I'd say lets not force rushed solutions. Once something is in it's difficult to take out and will just clutter the API.
kotlinx-io can take Okio as a reference implementation but might try a design closer to Java NIO.
Personally the NIO API is the cleanest I've seen.

**Comment by fzhinkin** [comment]
TL;DR: there were (and are) no signs that a proper stable FS API will be released soon (I mean in the next couple of months), so I decided to provide some API (FileSystem.list(Path): Collection<Path>) to make the library more usable.

**Comment by StefanOltmann** [comment]
I agree with this decision!

And it's unlikely that this API will change. Maybe the implementation, but `FileSystem.list(Path)` just makes sense.

**Comment by StefanOltmann** [comment]
Is it available using `SystemFileSystem.list(Path)`?
